Compiler warnings

Using my default settings, builds are verbose:

```sh
edd@rob:~/git/qtlbim(master)$ install.r                                                                  
* installing *source* package found in current working directory ...
* installing *source* package ‘qtlbim’ ...                                                                                                                                                                         files ‘demo/00Index’, ‘demo/qb.coda.tour.R’, ‘demo/qb.covar.tour.R’, ‘demo/qb.hyper.tour.R’, ‘demo/qb.mcmc.tour.R’, ‘demo/qb.plot.tour.R’, ‘demo/qb.scan.tour.R’, ‘demo/qb.sim.tour.R’, ‘demo/qb.tour.R’, ‘demo/qtl
bim.R’, ‘inst/doc/prototype.qtl.hyper.slides.Rnw’, ‘inst/doc/prototype.qtl.hyper.slides.pdf’, ‘inst/doc/qtlbim.Rnw’, ‘inst/doc/qtlbim.pdf’, ‘inst/external/hyper.paper.extra.Rnw’, ‘inst/external/hyper.slide.extra
.Rnw’, ‘inst/external/prototype.qtl.hyper.paper.Rnw’ are missing
file ‘DESCRIPTION’ has the wrong MD5 checksum
** using staged installation                                                                             
** libs
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c GlobalVars.c -o GlobalVars.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c GlobalVars_SingleTrait.c -o GlobalVars_SingleTrait.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c MatrixUtils.c -o MatrixUtils.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c MultipleTraitsMCMC.c -o MultipleTraitsMCMC.o
MultipleTraitsMCMC.c: In function ‘MT_QTLPOSITION’:                                                      
MultipleTraitsMCMC.c:835:7: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  835 |       if(UPDATEGENO==0) MT_Coefficient0(I,L,trait,QLOC[trait][L]);   
      |       ^~
MultipleTraitsMCMC.c:837:4: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  837 |    for(K=0;K<NC;K++) COEF[trait][I][L][K]=X[K];
      |    ^~~
MultipleTraitsMCMC.c: In function ‘QTLPOSITION_SameLocation’:
MultipleTraitsMCMC.c:924:7: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  924 |       if(UPDATEGENO==0) MT_Coefficient0(I,L,0,QLOC[0][L]);
      |       ^~
MultipleTraitsMCMC.c:925:4: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  925 |    for(K=0;K<NC;K++) COEF[0][I][L][K]=X[K];
      |    ^~~
MultipleTraitsMCMC.c: In function ‘multipleTraitsMCMC’:
MultipleTraitsMCMC.c:2938:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
 2938 |         if(DiffLocation==0)
      |         ^~
MultipleTraitsMCMC.c:2950:11: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
 2950 |           if(DiffLocation==1)
      |           ^~
MultipleTraitsMCMC.c:2440:24: warning: unused variable ‘S’ [-Wunused-variable]
 2440 | int NU=6; double H=0.1,S=2,TAU; //NU=degrees of freedom, H=heritability, TAU=scale
      |                        ^
MultipleTraitsMCMC.c:3437:12: warning: suggest parentheses around comparison in operand of ‘&’ [-Wparentheses]
 3437 | if(MULTIPLE>=1 & PH1 == 0) // only print sigma once per iteration.
      |    ~~~~~~~~^~~
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c RMultipleTraitsMCMCSetup.c -o RMultipleTraitsMCMCSetup.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c ROutputManager.c -o ROutputManager.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c RSingleTraitMCMCSetup.c -o RSingleTraitMCMCSetup.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c SingleTraitMCMC.c -o SingleTraitMCMC.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c SingleTraitMCMCSamplingRoutines.c -o SingleTraitMCMCSamplingRoutines.o
ccache gcc -I"/usr/share/R/include" -DNDEBUG     -fcommon -fpic  -g -O3 -Wall -pipe -pedantic  -std=gnu99 -c StatUtils.c -o StatUtils.o
StatUtils.c: In function ‘RANDOM’:
StatUtils.c:16:2: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   16 |  if(U>=1.0) U=1.0-(1e-10); if(U<=0.0) U=1e-10;
      |  ^~
StatUtils.c:16:28: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
   16 |  if(U>=1.0) U=1.0-(1e-10); if(U<=0.0) U=1e-10;
      |                            ^~
ccache gcc -Wl,-S -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o qtlbim.so GlobalVars.o GlobalVars_SingleTrait.o MatrixUtils.o MultipleTraitsMCMC.o RMultipleTraitsMCMCSetup.o ROutputManager.o RSingleTraitMCMCSetup.o SingleTraitMCMC.o SingleTraitMCMCSamplingRoutines.o StatUtils.o -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/00LOCK-qtlbim/00new/qtlbim/libs
** R
** data
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (qtlbim)
edd@rob:~/git/qtlbim(master)$ 
```

PR should be forthcoming in a few.

--- a/src/MultipleTraitsMCMC.c
+++ b/src/MultipleTraitsMCMC.c
@@ -821,30 +821,27 @@ if(MULTIPLE>=1)  PROB1 = LogLikelihood(Y,G,AMU,SIGMA);
 
     double S1=(PROB1-PROB0)+(PD10-PD1[L])+(PD2[L]-PD20); //Needed for genoupdate
          
- if(S1>log(RANDOM())) 
-	{
-		QLOC[trait][L]=QLNEW;
-   for(I=0;I<NS;I++)          
-		{
-			if(UPDATEGENO==1)
-			{
-				GENO[trait][I][L]=GENO1[I];
-				Coefficient(GENO1[I]);
-			}  
+    if(S1>log(RANDOM())) {
+            QLOC[trait][L]=QLNEW;
+            for(I=0;I<NS;I++) {
+                if(UPDATEGENO==1) {
+                    GENO[trait][I][L]=GENO1[I];
+                    Coefficient(GENO1[I]);
+                }
 
-      if(UPDATEGENO==0) MT_Coefficient0(I,L,trait,QLOC[trait][L]);
+                if(UPDATEGENO==0) MT_Coefficient0(I,L,trait,QLOC[trait][L]);
 
-			for(K=0;K<NC;K++) COEF[trait][I][L][K]=X[K];
+                for(K=0;K<NC;K++) COEF[trait][I][L][K]=X[K];
 
-      for(ph=0;ph<NPHENO;ph++)  GVALUE[ph][I]=G[ph][I];				
-  			
-		}		
+                for(ph=0;ph<NPHENO;ph++)  GVALUE[ph][I]=G[ph][I];
+
+            }		
 	 		
-		PD1[L]=PD10,PD2[L]=PD20;
-	}
-		   for(ph=0;ph<NPHENO;ph++)  free(G[ph]);
-		   free(G);
-	return;
+            PD1[L]=PD10,PD2[L]=PD20;
+    }
+    for(ph=0;ph<NPHENO;ph++)  free(G[ph]);
+    free(G);
+    return;
 }                              
 
 
@@ -916,15 +913,13 @@ if(MULTIPLE>=1)  PROB1 = LogLikelihood(Y,G,AMU,SIGMA);
 
     double S1=(PROB1-PROB0)+(PD10-PD1[L])+(PD2[L]-PD20); //Needed for genoupdate
          
- if(S1>log(RANDOM())) 
-	{
-    QLOC[0][L]=QLNEW;
-   for(I=0;I<NS;I++)          
-		{
-      if(UPDATEGENO==0) MT_Coefficient0(I,L,0,QLOC[0][L]);
-			for(K=0;K<NC;K++) COEF[0][I][L][K]=X[K];
+    if(S1>log(RANDOM())) {
+        QLOC[0][L]=QLNEW;
+        for(I=0;I<NS;I++) {
+            if(UPDATEGENO==0) MT_Coefficient0(I,L,0,QLOC[0][L]);
+            for(K=0;K<NC;K++) COEF[0][I][L][K]=X[K];
 
-    for(ph=0;ph<NPHENO;ph++)  GVALUE[ph][I]=G[ph][I];				
+            for(ph=0;ph<NPHENO;ph++)  GVALUE[ph][I]=G[ph][I];
       
 /*			if(UPDATEGENO==1)
 			{
@@ -932,13 +927,13 @@ if(MULTIPLE>=1)  PROB1 = LogLikelihood(Y,G,AMU,SIGMA);
 				Coefficient(GENO1[I]);
 			}  */
 			
-		}		
-	 	
-		PD1[L]=PD10,PD2[L]=PD20;
-	}
-		   for(ph=0;ph<NPHENO;ph++)  free(G[ph]);
-		   free(G);
-	return;
+        }
+
+        PD1[L]=PD10,PD2[L]=PD20;
+    }
+    for(ph=0;ph<NPHENO;ph++)  free(G[ph]);
+    free(G);
+    return;
 }                              
                      
 //*************************************************************
@@ -2437,7 +2432,7 @@ for(PH=0;PH<NPHENO;PH++)  MT_Mean(YBAR[PH],VP[PH],PH);
 
 if(CATEGORY==1 && MULTIPLE>=1) ResidualVariance_MultipleTraits();
 
-int NU=6; double H=0.1,S=2,TAU; //NU=degrees of freedom, H=heritability, TAU=scale
+int NU=6; double H=0.1/*,S=2*/,TAU; //NU=degrees of freedom, H=heritability, TAU=scale
 
 for(L=0;L<NQTL;L++)
     for(PH=0;PH<NPHENO;PH++)    
@@ -2936,40 +2931,36 @@ if(EPISTASIS==1)
 							}
           }
         if(DiffLocation==0)
-          for(PH=0;PH<NPHENO;PH++)
-          {			if(T!=0) MT_EpistasisIndicator_GROUP0(L1,L2,K1,K2,PH);
-      			    if(MULTIPLE==1)
-                  {
-                  for(PH1=0;PH1<NPHENO;PH1++){
-                   GAMMA[PH1][L1]=GAMMA[PH][L1];
-                   GAMMA[PH1][L2]=GAMMA[PH][L2];
-                   }
-                  PH=NPHENO-1;
-                  }
-          }
-          if(DiffLocation==1)
-  					for(PH=0;PH<NPHENO;PH++)
-            {		
-            	if(W_EPISTASIS!=0)
-              {
-								if(GAMMA[PH][L1]==0 && GAMMA[PH][L2]!=0) T=MT_SamplingOnePosition(L1,PH);
-								if(GAMMA[PH][L1]!=0 && GAMMA[PH][L2]==0) T=MT_SamplingOnePosition(L2,PH);
-								if(GAMMA[PH][L1]==0 && GAMMA[PH][L2]==0)
-								{
-									T=MT_SamplingOnePosition(L1,PH);
-									if(T!=0) T=MT_SamplingOnePosition(L2,PH);
-									if(CHRQTL[QCHR[PH][L1]]+2>CHR_NQTL[QCHR[PH][L1]]&&QCHR[PH][L1]==QCHR[PH][L2]
-					&&(GRID[QCHR[PH][L1]][QLOC[PH][L1]]-GRID[QCHR[PH][L2]][QLOC[PH][L2]])<=DQQ[QCHR[PH][L1]]) T=0;
+            for(PH=0;PH<NPHENO;PH++) {
+                if(T!=0) MT_EpistasisIndicator_GROUP0(L1,L2,K1,K2,PH);
+                if(MULTIPLE==1) {
+                    for(PH1=0;PH1<NPHENO;PH1++){
+                        GAMMA[PH1][L1]=GAMMA[PH][L1];
+                        GAMMA[PH1][L2]=GAMMA[PH][L2];
+                    }
+                    PH=NPHENO-1;
+                }
+            }
+        if(DiffLocation==1)
+            for(PH=0;PH<NPHENO;PH++) {
+            	if(W_EPISTASIS!=0) {
+                    if(GAMMA[PH][L1]==0 && GAMMA[PH][L2]!=0) T=MT_SamplingOnePosition(L1,PH);
+                    if(GAMMA[PH][L1]!=0 && GAMMA[PH][L2]==0) T=MT_SamplingOnePosition(L2,PH);
+                    if(GAMMA[PH][L1]==0 && GAMMA[PH][L2]==0) {
+                        T=MT_SamplingOnePosition(L1,PH);
+                        if(T!=0) T=MT_SamplingOnePosition(L2,PH);
+                        if(CHRQTL[QCHR[PH][L1]]+2>CHR_NQTL[QCHR[PH][L1]]&&QCHR[PH][L1]==QCHR[PH][L2]
+                           &&(GRID[QCHR[PH][L1]][QLOC[PH][L1]]-GRID[QCHR[PH][L2]][QLOC[PH][L2]])<=DQQ[QCHR[PH][L1]]) T=0;
 
-								}
-						  }		
+                    }
+                }
               	if(T!=0) MT_EpistasisIndicator_GROUP0(L1,L2,K1,K2,PH);
 
 							
-					  }		
+            }
 		      			
 
-       } 
+        }
 	if(GIBBS==0)
 	{
       			int TTT[NPHENO],sumTTT=0;
@@ -3434,7 +3425,7 @@ fprintf(File6,"\n");
  */
      
  
-if(MULTIPLE>=1 & PH1 == 0) // only print sigma once per iteration.
+if( (MULTIPLE>=1) & (PH1 == 0) ) // only print sigma once per iteration.
  { 
     for(I=0;I<NPHENO;I++)
           for(J=0;J<NPHENO;J++)           fprintf(File7,"%f\t",sigma[I][J]);

--- a/src/StatUtils.c
+++ b/src/StatUtils.c
@@ -13,7 +13,8 @@
 double RANDOM()
 {
 	double U=rand()/(RAND_MAX+1.0);
-	if(U>=1.0) U=1.0-(1e-10); if(U<=0.0) U=1e-10;
+	if(U>=1.0) U=1.0-(1e-10);
+        if(U<=0.0) U=1e-10;
 	return(U);
 }
 

